testbed/testbed/test_case.go

```diff
@@ -150,18 +150,20 @@ func (tc *TestCase) composeTestResultFileName(fileName string) string {
 	return fileName
 }
 
-// SetExpectedMaxCPU sets the percentage of one core agent is expected to consume at most.
-// Error is signalled if consumption during resourceCheckPeriod exceeds this number.
-// If this function is not called the CPU consumption does not affect the test result.
-func (tc *TestCase) SetExpectedMaxCPU(cpuPercentage uint32) {
-	tc.resourceSpec.expectedMaxCPU = cpuPercentage
-}
-
-// SetExpectedMaxRAM sets the maximum RAM in MiB the agent is expected to consume.
-// Error is signalled if consumption during resourceCheckPeriod exceeds this number.
-// If this function is not called the RAM consumption does not affect the test result.
-func (tc *TestCase) SetExpectedMaxRAM(ramMiB uint32) {
-	tc.resourceSpec.expectedMaxRAM = ramMiB
+// SetResourceLimits sets expected limits for resource consmption.
+// Error is signalled if consumption during ResourceCheckPeriod exceeds the limits.
+// Limits are modified only for non-zero fields of resourceSpec, all zero-value fields
+// fo resourceSpec are ignored and their previous values remain in effect.
+func (tc *TestCase) SetResourceLimits(resourceSpec ResourceSpec) {
```

Comment:
How about s/SetResourceLimits/ResourceLimits/ ?

Reply:
Relevant naming recommendation from Effective Go (see last sentence):
>Go doesn't provide automatic support for getters and setters. There's nothing wrong with providing getters and setters yourself, and it's often appropriate to do so, but it's neither idiomatic nor necessary to put Get into the getter's name. If you have a field called owner (lower case, unexported), the getter method should be called Owner (upper case, exported), not GetOwner. The use of upper-case names for export provides the hook to discriminate the field from the method. A setter function, if needed, will likely be called SetOwner.

I was following this recommendation in the codebase so far.

This is not exactly a pure setter function but very close to it.

Reply:
SG